cli/kubernetes/default_options_reader.go

```diff
@@ -0,0 +1,27 @@
+package kubernetes
+
+type DefaultOptionsReader struct{}
+
+// NewDefaultOptionsReader is a reader used by the Installer to
+// fill InstallationOptions with an initial default value, if such
+// is specified (dynamic per function, or static).
+func NewDefaultOptionsReader() DefaultOptionsReader {
+	return DefaultOptionsReader{}
+}
+
+// Read attempts to fill the option with a default, dynamic or static.
+func (reader DefaultOptionsReader) Read(option *InstallationOption) error {
+	// Give priority to a function which provides the default
+	// value dynamically.
+	if option.DynDefault != nil {
+		err := option.DynDefault(option)
+		if err != nil {
+			return err
+		}
+	} else if option.Default != nil {
+		option.Value = option.Default
+		option.Valid = true
```

Comment:
In the future we could have a  `Valid()` method on the InstallationOption that could have more complicated logic (e.g. checking that the system_domain actually looks like a domain). That function immediately return `true` if the Value is equal to the default one. Let's think about it when we need that validation logic.

Reply:
I would call such a function `Validate()`, and would have it call a `ValidateFunc` function field of `InstallationOption`, like we have a `DynDefaultFunc`. Agreed that this can be deferred to until after the demo.